api/envoy/config/route/v3/route_components.proto

```diff
@@ -837,6 +837,26 @@ message RouteAction {
   string prefix_rewrite = 5
       [(validate.rules).string = {well_known_regex: HTTP_HEADER_VALUE strict: false}];
 
+  // Indicates that location header needs to be rewritten. When upstream returns a 3XX status
+  // along with a location header, the location header will be overwritten with this value
+  // before being sent back as response to the client.
+  //
+  // This config has no effect if the upstream returns a status other than 301, 302, 303 and 307,
+  // or if a location header is not present in the response.
+  //
+  // Assume that there is an upstream cluster that returns a redirect for the below defined prefix
+  // and the returned location header contains /itm/not-found. The config below will rewrite the
+  // location header as /n/error.
+  //
+  //   .. code-block:: yaml
+  //
+  //     - match:
+  //         prefix: "/itm/"
+  //       route:
+  //         location_header_rewrite: "/n/error"
+  string location_header_rewrite = 36
```

Comment:
Do you think there will ever be a case to rewrite via regex, e.g. using `RegexMatchAndSubstitute`, or does it seem that a literal location always works?

I wonder if we want a more general `Rewriter` message that can be both string and `RegexMatchAndSubstitute` @envoyproxy/api-shepherds

Reply:
I was thinking about it too. Regex based rewrite will be more useful. Let me see how I can do that.

Reply:
+1 to have a wrapper message that contains an internal oneof with different rewrite types.

Reply:
location HTTP header rewrites are exactly what I need as well! However, could this be more generalized to support any HTTP header rewrite with regex?